lib/ansible/executor/task_executor.py

```diff
@@ -1044,9 +1047,22 @@ def _set_connection_options(self, variables, templar):
 
         # add extras if plugin supports them
         if getattr(self._connection, 'allow_extras', False):
-            for k in variables:
-                if k.startswith('ansible_%s_' % self._connection._load_name) and k not in options:
-                    options['_extras'][k] = templar.template(variables[k])
+            if self._connection.allow_extras is True:
+                display.deprecated(
+                    "Using the connection plugin's _load_name to determine extra vars is deprecated. "
+                    "Use the porting guide to update {self._connection_fqcn}'s allow_extras attribute "
+                    "from a boolean to a list regex pairs.",
+                    version="2.18"
+                )
+                match_extras = [(r'(^ansible_%s_)' % self._connection_fqcn.split('.')[-1], r'\1')]
+            else:
+                match_extras = self._connection.allow_extras
+
+            for extra_re, repl_re in match_extras:
```

Comment:
an exception for when they are not a list of tuples 'invalide plugin property, contact plugin author ...'

Reply:
I added some type checking to avoid unhandled tracebacks in the last commit